File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/ETertiaryFixedDimDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ETertiaryFixedDimDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd38
            AppTheme.STANDARD -> R.color.cstd38
            AppTheme.BLUE -> R.color.cbd38
            AppTheme.GREEN -> R.color.cgd38
            AppTheme.RED -> R.color.crd38
            AppTheme.NOTHING -> R.color.cnod38
        }
    }
}

File: lib/layer2_core/dark/e_tertiary_fixed_dim_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class ETertiaryFixedDimDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd38;
      case 'standard': return cstd38;
      case 'blue': return cbd38;
      case 'green': return cgd38;
      case 'red': return crd38;
      case 'nothing': return cnod38;
      default: return cstd38;
    }
  }
}
